### DISCUSSION TITLE
request: version information parameter on CLI

### DISCUSSION THREAD

**Discussion by Spreadcat** [Motion-Project/motion#1504] [General]
0. Reviewed guide and contributing documents? (Yes):
1. version : <= 4.31
2. installed as a package or compiled from sources: deb
3. standalone or part of third party: motion (standalone) 
4. video stream source: V4L+ rtsp
5. hardware x86: 
6. operating system: Raspian Buster

---

I've added a role to ansible galaxy to install and manage my motion installation on the raspberry.
Sine there are several parameter only available depending on the motion version, I have to query for that version after the package installation.
The current binary does not seem to have a parameter supporting this; usually `-v|--version` or something similar:

```
pi@raspberry:$ motion -v
motion: invalid option -- 'v'
motion Version 4.3.1, Copyright 2000-2019 Jeroen Vreeken/Folkert van Heusden/Kenneth Lavrsen/Motion-Project maintainers

Home page :	 https://motion-project.github.io/ 

usage:	motion [options]


Possible options:

-b			Run in background (daemon) mode.
-n			Run in non-daemon mode.
-s			Run in setup mode.
-c config		Full path and filename of config file.
-d level		Log level (1-9) (EMG, ALR, CRT, ERR, WRN, NTC, INF, DBG, ALL). default: 6 / NTC.
-k type			Type of log (COR, STR, ENC, NET, DBL, EVT, TRK, VID, ALL). default: ALL.
-p process_id_file	Full path and filename of process id file (pid file).
-l log file 		Full path and filename of log file.
-m			Disable motion detection at startup.
-h			Show this screen.

Motion is configured using a config file only. If none is supplied,
it will read motion.conf from current directory, ~/.motion or /etc/motion.
```

Issues:
a) The command takes about 3-7 seconds on my pi (depending on the hardware ofc) to execute.
b) The version information is available at the top of the help page, so I have to parse it from there (which works btw), but I'm not confident that the formatting of the output might not change in future versions.

Suggestion:
* add a parameter `-v|--version` that outputs only the version number of the current binary installation and immediately exists afterwards.

**Comment by tosiara** [comment]
So why is `motion -h | head -n 1` not an option?

**Comment by Spreadcat** [comment]
Hi,

thanks for the quick response. It is actually. The same way as I am currently using `motion -h | grep Version | cut -f3 -d ' ' | cut -f1 -d ','` to get that string.
But as mentioned: running the help takes a couple of seconds and that slows down the whole automation progress when retrieving the version number and b) a parameter like `-v|--version` might make it faster and easier to find. I was actually expecting it to be there from the beginning and stumbled about it first, when I tried to get the version number and run `motion -v` and got an error.
I've just waited with opening this issue until I had the ansible role done and was sure that I would invest enough time into motion in order to bother ;)

**Comment by tosiara** [comment]
I have checked and there is definitely a ~2s delay when running `-h`. Will check, why that happens
